src/pages/Campaign/CampaignButtonWithOptions.tsx

```diff
@@ -83,11 +83,27 @@ export default function CampaignButtonWithOptions({
       const encodedData = response.data.data.EncodedData
       try {
         await sendTransaction(rewardContractAddress, encodedData, BigNumber.from(0), transactionResponse => {
-          // TODO nguyenhuudungz: Compile a list of unclaimed rewards from `campaignLeaderboard`.
+          const accumulatedUnclaimedRewards = selectedCampaignLeaderboard.rewards
+            .filter(reward => !reward.claimed)
+            .reduce((acc: { [p: string]: CampaignLeaderboardReward }, value) => {
+              const key = value.token.chainId + '_' + value.token.address
+              if (acc[key] === undefined) {
+                acc[key] = value
+              } else {
+                acc[key] = {
+                  ...value,
+                  rewardAmount: value.rewardAmount.add(acc[key].rewardAmount),
+                }
+              }
+              return acc
+            }, {})
+          const rewardString = Object.values(accumulatedUnclaimedRewards)
+            .map(reward => reward.rewardAmount.toSignificant(DEFAULT_SIGNIFICANT) + ' ' + reward.token.symbol)
+            .join(' ' + t`and` + ' ')
```

Comment:
Give [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat/ListFormat) a try if it is better

Reply:
interesting bro but hard to read, not natural i think